wpimath/src/main/native/cpp/jni/WPIMathJNI_DARE.cpp

```diff
@@ -9,20 +9,20 @@
 #include <wpi/jni_util.h>
 
 #include "WPIMathJNI_Exceptions.h"
-#include "edu_wpi_first_math_WPIMathJNI.h"
+#include "edu_wpi_first_math_WPIMathJNI_DARE.h"
 #include "frc/DARE.h"
 
 using namespace wpi::java;
 
 extern "C" {
 
 /*
- * Class:     edu_wpi_first_math_WPIMathJNI
+ * Class:     edu_wpi_first_math_WPIMathJNI_00024DARE
  * Method:    dareDetailABQR
  * Signature: ([D[D[D[DII[D)V
  */
 JNIEXPORT void JNICALL
-Java_edu_wpi_first_math_WPIMathJNI_dareDetailABQR
+Java_edu_wpi_first_math_WPIMathJNI_00024DARE_dareDetailABQR
```

Comment:
Yuck.  Are we *sure* we want to use inner classes to do this?

Reply:
For a JNI function like Eigen's exp(), there's three options:

1. No prefix: `WPIMathJNI.exp()` (status quo that has a name collision with Pose3d's exp())
2. Prefix: `WPIMathJNI.eigenPow()`
3. static class: `WPIMathJNI.Eigen.pow()`

Option 1 is the status quo that has a name collision with Pose3d's exp(); that's what we want to avoid. Options 2 and 3 both solve the ambiguity issue, but option 3 is cleaner at the user callsite.

The `_00024` thing is just how Java encodes the `$` character of the internal class in its JNI name mangling. It doesn't show up in the actual user call.

Reply:
I like option 4 as well.